test/test_browser.py

```diff
@@ -5637,3 +5637,26 @@ def test_emrun(self):
       self.assertContained('Testing ASCII characters: !"$%&\'()*+,-./:;<=>?@[\\]^_`{|}~', stdout)
       self.assertContained('Testing char sequences: %20%21 &auml;', stdout)
       self.assertContained('hello, error stream!', stderr)
+
+  def test_worker_error_postmessage(self):
+    create_file('src.cpp', r'''
+      #include <thread>
+
+      void crash() {
+        int (*fun)(void) = nullptr;
+        (fun)();
+      }
+
+      int main() {
+        std::thread worker(crash);
+        worker.join();
+        return 0;
+      }
+      ''')
+    create_file('pre.js', r'''
+      console.log = function(message) {
+        reportResultToServer(message);
```

Comment:
I would hope the result here would an exception on the main thread so we can detect that rather than detecting the log message.    Something like `run_browser(..., expected='exception:xxx'`).